### ISSUE TITLE
(Self-Voicing) Gesture and Touch - repeat responses

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab-basics#255]
**When moving mass spheres**
It seems when I move and do a quick release (gesture) or lift my finger (touch) I often get back to back object and context responses with the latest one not interrupting the stale one. 

Also when I do a slow release, like I pause before I release, I feel like I can easily get a duplicate response. That is as I release the sphere (lift my finger) the same object and context responses that i just heard fire again.

I think there may be some improvement we can make to the timing of responses in coordination with the gestures/touch, but I am not sure how to guide the changes.

**For regular touch**, my feeling is that when a move happens quickly, the latest info should over ride the stale information. This means in our originally pattern, we delivered information at the end of the drag.

I have a feeling we should still be doing it that way, and it's possible we have muddied the waters a little bit as we shifted focus towards non-visual access.

**For Custom Gesture**
However, with custom gesture and non-visual access, it is important to get that starting information and the final latest information, and ideally with little to no repetition.

Currently, this is what I am experiencing:
1. We get initial name and object value on focus. 
  - _This is good. Keep it this way._
2. We also get object name and starting value of object after "Grabbed". 
  - _I am not sure we need this repeated at start of grabbed mode, but I think it's ok to keep it that way for now._
3. If I keep my finger down and do not move the sphere, the name and distance (object response) keep repeating. 
  - _Can we stop this repetition from happening? If no change in object value is happening (even if I have my finger down), I don't think I should be getting object or context responses._ @jessegreenberg maybe movement is being detecting by the model even if it is less than 0.1 kilometers? Maybe that's the issue? Anyways, if there is no meaningful change in distance it shouldn't repeat after it has been announced once.
4. Upon a release I hear "Released", and I often hear again, the object response (distance value) and the context response (force change). I hear both and often when there is actually no change in distance and or force after the end of my drag and upon release. 
  - _Is it possible to not repeat the the object and context response after release if there is no actual distance change happening when I lift my finger?_

@jessegreenberg, this is the repetition i was feeling uneasy about in yesterday's meeting. I think you have tweaks to timing of responses on your list and you will address them for the haptics interviews once you have everything else done.

**Comment by jessegreenberg** [comment]
> If I keep my finger down and do not move the sphere, the name and distance (object response) keep repeating.

This part was fixed in the above commit, I am hopeful it will improve repetition in many of the cases @terracoda listed above. I am going to try to fix #253 before publishing a version for review.

**Comment by jessegreenberg** [comment]
@terracoda here is a version with this change, can you please review the self-voicing while dragging with gesture?

https://phet-dev.colorado.edu/html/gravity-force-lab-basics/1.1.0-dev.29/phet/gravity-force-lab-basics_en_phet.html?supportsGestureControl&supportsSelfVoicing

**Comment by terracoda** [comment]
Everything fixed for regular touch input with Self-Voicing.
Everything fixed for custom gesture input with Self-Voicing.
Nice work.
I found a weird focus issue when using keyboard with Self-Voicing. I'll create a new issue for that.

**Comment by terracoda** [comment]
@jessegreenberg, I think it is safe to close this issue from my perspective, though I haven't tested in Test Flight yet.

**Comment by terracoda** [comment]
Assigning to you for closing :-)

**Comment by jessegreenberg** [comment]
Great, ok thanks for checking!

> I found a weird focus issue when using keyboard with Self-Voicing. I'll create a new issue for that.

Sounds good, thanks.

Closing.

**Comment by jessegreenberg** [comment]
> though I haven't tested in Test Flight yet.

@terracoda, no worries, it won't be fixed in TestFlight until I publish a new version of the app. But if it is fixed in the sim it should be fixed there too.
